keps/sig-cli/4944-kustomize-wasm-plugin/README.md

```diff
@@ -0,0 +1,878 @@
+distributing and managing binaries for each environment.
+
+#### Story 2
+
+A platform team wants to share team-specific resource transformation logic as
+Kustomize plugins. By distributing Wasm plugins through OCI registries, team
+members can easily obtain plugins and start using them without worrying about
+execution environment setup or binary compatibility.
+
+#### Story 3
+
+An application developer wants to use a popular open-source Kustomize plugin
+that implements advanced JSON patching, which is distributed through an OCI
+registry. Since the plugin is implemented in WebAssembly and runs in a sandboxed
+environment, they can confidently use the third-party plugin without worrying
+about security vulnerabilities or system access. The developer can easily fetch
+the latest version of the plugin from the OCI registry, verify its integrity,
+and run it safely within their build pipeline, knowing that the plugin's
+capabilities are limited to the intended resource transformations.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+- Current kustomize plugin systen that called `Exec/Container KRM functions` is
+  using stdin/stdout to communicate with kustomize to plugins.
+  - WASI is support to stdin/stdout, so We reuse the current interface
+    definitions and users can migrate with minimal changes.
+- Now,
+  [Wasm binary can be distributed with OCI registry](https://tag-runtime.cncf.io/wgs/wasm/deliverables/wasm-oci-artifact/).
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+1. WASI Specific is not yet GA. Any disruptive changes in the future will incur
+   user transition costs.
+1. The Wasm OCI Artifact specification is a relatively new standard in the OCI
+   ecosystem. Maybe not mature enough.
+1. Containaized KRM function is alpha feature but that used some user.
+   - Provide container to exec conversion tools.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Implement with [wazero](https://github.com/tetratelabs/wazero) to WASM runtime
+like the
+[kube-scheduler-wasm-extension](https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension)
+project.
```

Comment:
```suggestion
The implementation will use [wazero](https://github.com/tetratelabs/wazero) library for WASM runtime support. This is the same library used by [kube-scheduler-wasm-extension](https://github.com/kubernetes-sigs/kube-scheduler-wasm-extension) project.
```

Reply:
Additional question, there are several other libraries providing wasm support. Wasmer, Wasmtime and there's also native Go support. Can you describe in a few sentences why this, and not the other ones?